assets/services/auth-service.js

```diff
@@ -1,76 +1,52 @@
 angular.module('beamer.auth', ['beamer.session'])
 
-.service('authService', ['sessionService', '$http', '$location',
+.service('auth', ['$http', 'session', '$q',
+    function($http, session, $q) {
 
+        var root = 'http://localhost:1337';
 
-    function($http, $location, sessionService) {
-
+        /** Checks session expiry. Returns boolean*/
+        this.isAuthenticated = function() {
+            return session.isValid();
+        };
 
+        /** Posts user to /signup and creates a session if 
+         *server responds with a token. Returns a promise.
+         */
         this.createNewUser = function(user) {
-            // // send user info to server
-            // $http.post('http://localhost:1337/user/signup', user)
-            //     .success(SignupSuccess)
-            //     .error(SignupError);
-            console.log(sessionService.startSession())
-
-        }
+            var promise = $http({
+                method: 'POST',
+                url: root + '/user/signup',
+                data: user
+            }).then(function(data) {
+                if (data.data.token) {
+                    session.create(data.data.token);
+                    return data;
+                }
+            });
+            return promise;
+        };
 
+        /** Posts user to /login and creates a session if 
+         *server responds with a token. Returns a promise.
```

Comment:
Not going to reject, because it's really up to you how thorough you want to be - but you should use the jsdocs tags

@param
@return {<type>} <name> <description>